sw/device/tests/aes_sideload_test.c

```diff
@@ -0,0 +1,290 @@
+  };
+
+  LOG_INFO("Encrypting the plain text with the side loaded key.");
+  CHECK_DIF_OK(dif_aes_start(&aes, &transaction, NULL, NULL));
+  dif_aes_data_t in_data_plain;
+  memcpy(in_data_plain.data, kPlainText, sizeof(kPlainText));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusInputReady, true, TIMEOUT);
+  CHECK_DIF_OK(dif_aes_load_data(&aes, in_data_plain));
+
+  // Trigger the AES encryption and wait for it to complete.
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerStart));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusOutputValid, true, TIMEOUT);
+
+  //  Verify that the ciphertext is different from the plaintext.
+  dif_aes_data_t out_data;
+  CHECK_DIF_OK(dif_aes_read_output(&aes, &out_data));
+  CHECK_ARRAYS_NE(out_data.data, kPlainText, ARRAYSIZE(kPlainText));
+
+  // Finish the ECB encryption transaction.
+  CHECK_DIF_OK(dif_aes_end(&aes));
+
+  LOG_INFO("Decrypting the cypher text with the side loaded key.");
+  transaction.operation = kDifAesOperationDecrypt;
+  CHECK_DIF_OK(dif_aes_start(&aes, &transaction, NULL, NULL));
+  memcpy(in_data_plain.data, out_data.data, sizeof(out_data.data));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusInputReady, true, TIMEOUT);
+  CHECK_DIF_OK(dif_aes_load_data(&aes, in_data_plain));
+
+  // Trigger the AES decryption and wait for it to complete.
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerStart));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusOutputValid, true, TIMEOUT);
+
+  // Verify that the output is equal to the plain text.
+  dif_aes_data_t out_data_plain;
+  CHECK_DIF_OK(dif_aes_read_output(&aes, &out_data_plain));
+  CHECK_ARRAYS_EQ(out_data_plain.data, kPlainText, ARRAYSIZE(kPlainText));
+
+  // Finish the ECB decryption transaction.
+  CHECK_DIF_OK(dif_aes_end(&aes));
+
+  LOG_INFO("Clearing the side loaded key.");
+
+  // Clear the key in the keymgr and decrypt the ciphertext again.
+  CHECK_DIF_OK(
+      dif_keymgr_sideload_clear_set_enabled(&keymgr, kDifToggleEnabled));
+  CHECK_DIF_OK(dif_aes_start(&aes, &transaction, NULL, NULL));
+
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusInputReady, true, TIMEOUT);
+  CHECK_DIF_OK(dif_aes_load_data(&aes, in_data_plain));
+
+  // Trigger the AES decryption and wait for it to complete.
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerStart));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusOutputValid, true, TIMEOUT);
+
+  // Verify that output is not equal to the plain text.
+  CHECK_DIF_OK(dif_aes_read_output(&aes, &out_data_plain));
+  CHECK_ARRAYS_NE(out_data_plain.data, kPlainText, ARRAYSIZE(kPlainText));
+  CHECK_DIF_OK(dif_aes_end(&aes));
+}
+
+static void init_kmac_for_keymgr(void) {
+  dif_kmac_t kmac;
+  CHECK_DIF_OK(
+      dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  // Configure KMAC hardware using software entropy.
+  CHECK_DIF_OK(dif_kmac_configure(&kmac, (dif_kmac_config_t){
+                                             .sideload = true,
```

Comment:
This isn't needed when this issue https://github.com/lowRISC/opentitan/issues/13594 is fixed

we'd better configure the prefix with correct value. 
https://github.com/lowRISC/opentitan/blob/85bb174a344bdc725fecb981c370c1af799a61a7/sw/device/tests/sim_dv/keymgr_key_derivation_test.c#L143-L146

another general suggestion: if you find a function is reusable, consider adding it in *_testutils.c so that it's easy for future maintenance.